File: src/main/java/com/deneb/client/command/commands/Vanish.java
package com.deneb.client.command.commands;

import com.deneb.client.command.Command;
import com.deneb.client.features.ModuleManager;
import com.deneb.client.utils.ChatUtil;

/**
 * Created by B_312 on 12/12/20
 * Updated by B_312 on 01/15/21
 */
@Command.Info(command = "vanish", description = "Vanish riding entity.")
public class Vanish extends Command {

    @Override
    public void onCall(String s, String[] args) {
        if (args.length == 0) {
            ChatUtil.sendNoSpamMessage(getSyntax());
            return;
        }

        try {
            String key = args[0];
            if (key.equalsIgnoreCase("dismount")) ModuleManager.getModuleByName("EntityDeSync").enable();
            if (key.equalsIgnoreCase("remount")) ModuleManager.getModuleByName("EntityDeSync").disable();
        } catch (Exception e) {
            ChatUtil.sendNoSpamErrorMessage(getSyntax());
        }
    }

    @Override
    public String getSyntax() {
        return "vanish <dismount/remount>";
    }

}

File: src/main/java/club/deneb/client/command/commands/Vanish.kt
package club.deneb.client.command.commands

import club.deneb.client.command.Command
import club.deneb.client.features.ModuleManager
import club.deneb.client.features.modules.misc.EntityDeSync
import club.deneb.client.utils.ChatUtil.sendNoSpamErrorMessage
import club.deneb.client.utils.ChatUtil.sendNoSpamMessage

/**
 * Created by B_312 on 12/12/20
 * Updated by B_312 on 01/15/21
 */
@Command.Info(command = "vanish", description = "Vanish riding entity.")
class Vanish : Command() {
    override fun onCall(s: String, vararg args: String) {
        if (args.isEmpty()) {
            sendNoSpamMessage(getSyntax())
            return
        }
        try {
            val key = args[0]
            if (key.equals("dismount", ignoreCase = true)) ModuleManager.getModule(EntityDeSync::class.java).enable()
            if (key.equals("remount", ignoreCase = true)) ModuleManager.getModule(EntityDeSync::class.java).disable()
        } catch (e: Exception) {
            sendNoSpamErrorMessage(getSyntax())
        }
    }

    override fun getSyntax(): String {
        return "vanish <dismount/remount>"
    }
}